File: LeetCode/Java/88-two-pointers-合并两个有序数组.java
package LeetCode.Java;
/*
 * @lc app=leetcode.cn id=88 lang=java
 *
 * [88] 合并两个有序数组
 */

// @lc code=start
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        if (n == 0)
            return;
        int p = m + n - 1;
        int t1 = m - 1, t2 = n - 1;
        while (t1 >= 0 && t2 >= 0) {
            nums1[p--] = (nums1[t1] > nums2[t2] ? nums1[t1--] : nums2[t2--]);
        }
        while (t2 >= 0) {
            nums1[p--] = nums2[t2--];
        }
    }
}
// @lc code=end

File: LeetCode/Python/88-two-pointers-合并两个有序数组.py
#
# @lc app=leetcode.cn id=88 lang=python3
#
# [88] 合并两个有序数组
#
from typing import List


# @lc code=start
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        i, j, p = m - 1, n - 1, m + n - 1
        while i >= 0 and j >= 0:
            if nums1[i] > nums2[j]:
                nums1[p] = nums1[i]
                i -= 1
            else:
                nums1[p] = nums2[j]
                j -= 1
            p -= 1
        while j >= 0:
            nums1[p] = nums2[j]
            p -= 1
            j -= 1


# @lc code=end
